### ISSUE TITLE
Project Inspect confusion

### ISSUE THREAD

**Issue by OmarAlkassab** [platformio/platform-espressif32#247]
I compiled my ESP32 code in PlatformIO using VSCode, and get the following results:
![2019-11-12 13_34_28-PIO Home - work (Workspace) - Visual Studio Code](https://user-images.githubusercontent.com/31249807/68673801-3bb47100-055d-11ea-98db-21a6ae9a0e4d.png)

Then I used the Project Inspect function and get the following result:
![2019-11-12 14_57_49-PIO Home - work (Workspace) - Visual Studio Code](https://user-images.githubusercontent.com/31249807/68673843-4c64e700-055d-11ea-8aec-687eee9a41eb.png)

The first one shows that my code uses 14.7% of the RAM, but the second one shows 89% !
Which of them is correct?

**Comment by valeros** [comment]
Hi @OmarAlkassab! Thanks for the report. Indeed, the results are different because we use different mechanisms to calculate memory utilization. ESP32 platform is quite tricky in terms of application layout and if I'm not mistaken, ESP32 has three different RAM types and some of them may contain executable data. So I'd say that the report from Project Inspector is a bit more precise. In any case, could you add a buffer of size 50kb to your application and try to build your project? Does the linker report a memory overflow?

**Comment by OmarAlkassab** [comment]
I tried to add `byte buffer[50*1024] = {1};` but nothing happen, there is no difference nor in Project Inspect, nor in the compiling process!

**Comment by OmarAlkassab** [comment]
I tried `byte buffer[50*1024];` and in the `loop()` function, I added `buffer[0]=1;`, the linker didn't show any overflow, but when I tried `byte buffer[500*1024];` the linker show an overflow.
![2019-11-20 16_47_55-main cpp - work (Workspace) - Visual Studio Code](https://user-images.githubusercontent.com/31249807/69248765-8a918480-0bb5-11ea-93b7-49a4a1eb43c2.png)

**Comment by OmarAlkassab** [comment]
Trying `byte buffer[60*1024];` gives the following results:
![2019-11-20 17_15_49-PIO Home - work (Workspace) - Visual Studio Code](https://user-images.githubusercontent.com/31249807/69251135-9f701700-0bb9-11ea-9f0a-226cb98e2e61.png)

![2019-11-20 17_16_27-PIO Home - work (Workspace) - Visual Studio Code](https://user-images.githubusercontent.com/31249807/69251171-ad259c80-0bb9-11ea-94a2-02e94cfe7114.png)

**Comment by valeros** [comment]
Could you send us your project for analysis?

**Comment by OmarAlkassab** [comment]
Unfortunately no, because the project is team work, and contains many confidential concepts and innovative ideas.
If you cannot help me without seeing the code, no problem, many thanks.

**Comment by 84berlinetta** [comment]
Does anyone have an answer for Omar's question?  I have encountered the same issue, and I am wondering which analysis of resource utilization is correct.  I have noticed that if I compile a project which exceeds the 100% indication in the inspector statistics screen, I experience issues with the run-time operation which would indicate the design is memory starved.

BTW - I am in the same position as Omar, where I cannot release the project for scrutiny.  However, I would suspect that someone could exercise one of the example ESP32 WiFi projects to investigate the differences between the resources reported by the linker vs. the inspector.
